src/CosmosClientOptions.ts

```diff
@@ -31,4 +38,8 @@ export interface CosmosClientOptions {
    * Use an agent such as https://github.com/TooTallNate/node-proxy-agent if you need to connect to Cosmos via a proxy
    */
   agent?: Agent;
+  /** An optional function to be called on each request
+   * // TODO Add docs
+   */
+  executeRequest?: ExecuteRequest;
```

Comment:
@christopheranderson I'm definitely not set on this API. I would like to maybe rethink it when I do the pipeline work but for now, I need something here to get this into the portal

Reply:
Why do we need this executeRequest thing?

Reply:
Is this just needed for the portal? Could we do it in a hacky way where we expose it as a public type on ClientContext so you can override it?